src/events/EventsSection.tsx

```diff
@@ -122,60 +230,269 @@ export const EventsSection = () => {
-                  </ToolbarItem>
-                </>
-              }
-              columns={[
-                {
-                  name: "time",
-                  displayKey: "events:time",
-                  cellRenderer: (row) => moment(row.time).format("LLL"),
-                  cellFormatters: [expandable],
-                },
-                {
-                  name: "userId",
-                  displayKey: "events:user",
-                  cellRenderer: UserDetailLink,
-                },
-                {
-                  name: "type",
-                  displayKey: "events:eventType",
-                  cellRenderer: StatusRow,
-                },
-                {
-                  name: "ipAddress",
-                  displayKey: "events:ipAddress",
-                  transforms: [cellWidth(10)],
-                },
-                {
-                  name: "clientId",
-                  displayKey: "events:client",
-                },
-              ]}
-              emptyState={
-                <ListEmptyState
-                  message={t("emptyEvents")}
-                  instructions={t("emptyEventsInstructions")}
-                />
-              }
-            />
+            <Flex>
+              <FlexItem>
+                <Dropdown
+                  id="user-events-search-select"
+                  data-testid="UserEventsSearchSelector"
+                  toggle={
+                    <DropdownToggle
+                      data-testid="userEventsSearchSelectorToggle"
+                      onToggle={onDropdownToggle}
+                      className="keycloak__user_events_search_selector_dropdown__toggle pf-u-mt-md pf-u-ml-md pf-u-mb-md"
+                    >
+                      {t("searchForEvent")}
+                    </DropdownToggle>
+                  }
+                  isOpen={searchDropdownOpen}
+                >
+                  <Form
+                    isHorizontal
+                    className="keycloak__user_events_search__form"
+                    data-testid="searchForm"
+                  >
+                    <FormGroup
+                      label={t("userId")}
+                      fieldId="kc-userId"
+                      className="keycloak__user_events_search__form_label"
+                    >
+                      <TextInput
+                        ref={register()}
+                        type="text"
+                        id="kc-userId"
+                        name="userId"
+                        data-testid="userId-searchField"
+                        onChange={(_val, event) => {
+                          const { value } = event.currentTarget;
+                          setValue("userId", value);
+                          setIsDirty(true);
+                        }}
+                        defaultValue={
+                          formValues?.userId ? formValues?.userId : ""
+                        }
```

Comment:
The size of this statement can be reduced.
```suggestion
                        defaultValue={formValues?.userId ?? ""}
```